sklearn/model_selection/_search.py

```diff
@@ -1804,3 +1807,351 @@ def _run_search(self, evaluate_candidates):
+                alternative="two-sided",
+                zero_method="zsplit",  # Fallback condition in the case of ties
+            )[1]
+            for pair in tests
+        ]
+
+        # Determine the highest performing model that is not significantly
+        # different from the best average score column
+        h_cutoff = np.nanmax(
+            self._cv_means[
+                [
+                    pair[0]
+                    for pair in tests
+                    if pair[0] != self._best_score_idx
+                    and pvals[tests.index(pair)] > self._alpha
+                ]
+            ]
+        )
+
+        l_cutoff = np.nanmin(
+            self._cv_means[
+                [
+                    pair[0]
+                    for pair in tests
+                    if pair[0] != self._best_score_idx
+                    and pvals[tests.index(pair)] > self._alpha
+                ]
+            ]
+        )
+
+        return l_cutoff, h_cutoff
+
+    def call_percentile(self):
+        """
+        Returns the simplest model whose performance is within the `eta`
+        percentile of the average highest performing model.
+        """
+
+        # Estimate the indicated percentile, and its inverse, across folds for
+        # each column of the grid
+        perc_cutoff = np.nanpercentile(
+            self._score_grid, [100 * self._eta, 100 - 100 * self._eta], axis=1
+        )
+
+        # Determine bounds of the percentile interval
+        h_cutoff = perc_cutoff[0, self._best_score_idx]
+        l_cutoff = perc_cutoff[1, self._best_score_idx]
+
+        return l_cutoff, h_cutoff
+
+    @staticmethod
+    def simplify(param, rule="se", sigma=1, eta=0.68, alpha=0.01):
+        """
+        Callable to be run as `refit` argument of `GridsearchCV` or `RandomSearchCV`.
+
+        Parameters
+        ----------
+        param : str
+            Parameter with the largest influence on model complexity.
+        sigma : int
+            Number of standard errors tolerance in the case that a standard
+            error threshold is used to filter outlying scores across folds.
+            Only applicable if `rule`=='se'. Default is 1.
+        eta : float
+            Acceptable percent tolerance in the case that a percentile
+            threshold is used. Only applicable if `rule`=='percentile'.
+            Default is 0.68.
+        alpha : float
+            Alpha-level to use for signed wilcoxon rank sum testing.
+            Only applicable if `rule`=='ranksum'. Default is 0.01.
+
+        """
+        from functools import partial
+
+        def razor_pass(cv_results_, param, rule, sigma, alpha, eta):
+            rcv = Razors(
+                cv_results_,
+                param,
+                rule=rule,
+                sigma=sigma,
+                alpha=alpha,
+                eta=eta,
+            )
+            return rcv._best_low_complexity()
+
+        return partial(
+            razor_pass,
```

Comment:
Good catch!

Reply:
As a start, I've at least gone ahead and at least defined the closure outside of the parent function. It's actually rather tricky here since the function `partial` approach is critical to making this logic work. I haven't retested it yet for pickeability, but please let me know whether this will suffice as is or whether you can think of a better way of addressing this issue.